pkg/util/instancemetadata/dev.go

```diff
@@ -25,5 +27,6 @@ func newDev() (InstanceMetadata, error) {
 		subscriptionID: os.Getenv("AZURE_SUBSCRIPTION_ID"),
 		location:       os.Getenv("LOCATION"),
 		resourceGroup:  os.Getenv("RESOURCEGROUP"),
+		environment:    &azure.PublicCloud,
```

Comment:
check AZURE_ENVIRONMENT: if set, call `azure.EnvironmentFromName()`; if not default to `&azure.PublicCloud`.  We'll be wanting to run the RP from our latops against test environments in other clouds before you know it.